packages/mask/src/social-network-adaptor/mirror.xyz/collecting/identity.ts

```diff
@@ -0,0 +1,86 @@
+import { delay } from '@dimensiondev/kit'
+import { EnhanceableSite, ProfileIdentifier } from '@masknet/shared-base'
+import { Mirror, Writer } from '@masknet/web3-providers'
+import type { SocialNetworkUI } from '../../../social-network/types'
+import { creator } from '../../../social-network/utils'
+import { mirrorBase } from '../base'
+
+const getCurrentUserInfo = async () => {
+    if (location.host !== EnhanceableSite.Mirror) return
+    const userAddress = localStorage.getItem('mirror.userAddress') as string | null
+
+    if (!userAddress) return
+    return Mirror.getWriter(userAddress)
+}
+
+function resolveLastRecognizedIdentityInner(
+    ref: SocialNetworkUI.CollectingCapabilities.IdentityResolveProvider['recognized'],
+    cancel: AbortSignal,
+) {
+    const assign = async () => {
+        await delay(2000)
+
+        const writer = await getCurrentUserInfo()
+        if (!writer) return
+
+        ref.value = {
+            avatar: writer.avatarURL,
+            nickname: writer.displayName,
+            identifier: ProfileIdentifier.of(mirrorBase.networkIdentifier, writer.address).unwrapOr(undefined),
+        }
+    }
+
+    assign()
+
+    window.addEventListener('locationchange', assign, { signal: cancel })
+}
+
+function resolveCurrentVisitingIdentityInner(
+    ref: SocialNetworkUI.CollectingCapabilities.IdentityResolveProvider['recognized'],
+    cancel: AbortSignal,
+) {
+    const assign = async () => {
+        const script = document.getElementById('__NEXT_DATA__')?.innerHTML
+        if (!script) return
+        const INIT_DATA = JSON.parse(script)
+        if (!INIT_DATA) return
+        const writer = INIT_DATA.props?.pageProps?.project as Writer
+        if (!writer) {
+            if (location.pathname !== '/dashboard') return
+
+            // when current page is dashboard
+            const currentUser = await getCurrentUserInfo()
+            if (!currentUser) return
+            ref.value = {
+                avatar: currentUser?.avatarURL,
+                nickname: currentUser?.displayName,
+                identifier: ProfileIdentifier.of(mirrorBase.networkIdentifier, currentUser.address).unwrapOr(undefined),
+            }
```

Comment:
Did you forget `return` here?
```suggestion
            }
            return
```